app/Main.hs

```diff
@@ -122,17 +122,19 @@ main = do
       exitFailure
     Right C.Config {..} -> do
       chan <- newTChanIO
+      let task rf = Task { taskDeployPath    = configDeployPath
+                         , taskRepository    = configRepo
+                         , taskRevision      = configRevision
+                         , taskReleaseFormat = rf }
       let printFnc dest str = atomically $
             writeTChan chan (PrintMsg dest str)
           hap sshOpts =  do
             r <- Hap.runHapistrano sshOpts printFnc $
               case optsCommand of
                 Deploy releaseFormat n -> do
-                  release <- Hap.pushRelease Task
-                    { taskDeployPath    = configDeployPath
-                    , taskRepository    = configRepo
-                    , taskRevision      = configRevision
-                    , taskReleaseFormat = releaseFormat }
+                  release <- case configVcAction of
```

Comment:
@juanpaucar Thanks for the feedback.

Which function is long - Do you mean the `main` function itself ?

One another way of doing this is making `pushRelease` accept the parameter `bool` like you said. But IMO that is messy as we need to check for the bool and do some actions and ignore some actions based on that. Also since `pushRelease` function is exported, it is a backward incompatible change (if that matters.). I felt it is easy to have separate functions to make it easier for readability and maintenance in the long term.